trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisLocateSingleHopTradeRouteRepository.java

```diff
@@ -0,0 +1,118 @@
+package io.edpn.backend.trade.adapter.persistence.repository;
+
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisSingleHopEntity;
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisStationEntity;
+import io.edpn.backend.trade.adapter.persistence.entity.MybatisValidatedCommodityEntity;
+import io.edpn.backend.trade.adapter.persistence.filter.MybatisLocateSingleHopTradeFilter;
+import org.apache.ibatis.annotations.One;
+import org.apache.ibatis.annotations.Result;
+import org.apache.ibatis.annotations.Results;
+import org.apache.ibatis.annotations.Select;
+
+import java.util.List;
+
+public interface MybatisLocateSingleHopTradeRouteRepository {
+    
+    @Select("""
+            <script>
+            WITH stations_in_range AS (SELECT station.id,
+                                              system_id,
+                                              arrival_distance
+                                       FROM public.station
+                                                INNER JOIN system ON station.system_id = system.id
+                                       WHERE st_3ddwithin(st_makepoint(#{xCoordinate}, #{yCoordinate}, #{zCoordinate}), coordinates_geom, 300)
+                                        <if test='!includeFleetCarriers'>
+                                         AND station.name NOT SIMILAR TO '[A-Za-z0-9]{3}-[A-Za-z0-9]{3}'
```

Comment:
I guess this is a workable solution, but wouldn't a boolean flag (or something) for fleet carriers be better? Not only, because it's more "correct", but also for performance? (I'm guessing there's a function index for this, but indices aren't "free", either.)

Reply:
We already have a column for it.
IMC isn't up and running, this removes all the senseless data and will be dropped when that's done. Noted in PR comments.